src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -108,7 +102,17 @@ public function process(ContainerBuilder $container)
                 throw new InvalidArgumentException(sprintf('Service "%s" must implement interface "%s".', $id, $interface));
             }
 
-            $definition->addMethodCall('addSubscriberService', array($id, $class));
+            foreach ($class::getSubscribedEvents() as $eventName => $params) {
```

Comment:
this requires adding a ClassResource for the event subscriber class to invalidate the container when changing the class code. Otherwise, adding a new event in a subscriber would require clearing the cache manually in dev